datadog/internal/validators/validators.go

```diff
@@ -232,6 +232,29 @@ func ValidateAWSAccountID(v any, p cty.Path) diag.Diagnostics {
 	return diags
 }
 
+var _ schema.SchemaValidateDiagFunc = ValidateBasicEmail
+var basicEmailRe = regexp.MustCompile("^[^@]+@[^@]+\\.[^@.]+$")
```

Comment:
FYI this regex catch some special char like carriage return https://regex101.com/r/yGRf01/1

Reply:
My goal here is only to catch simple usage errors, with a very basic regex.
The full restrictive email regex is a lot more [hairy](https://stackoverflow.com/a/201378), and doesn't really bring more value since the only reliable way to test an email address is to send something to it.